Application/ControlMsrE2/ControlMsrE2UI.c

```diff
@@ -0,0 +1,277 @@
+/** @file
+ Control MSR 0xE2 UI Input/Output in the widest sense
+ 
+ Copyright (c) 2020, Brumbaer. All rights reserved.<BR>
+ This program and the accompanying materials
+ are licensed and made available under the terms and conditions of the BSD License
+ which accompanies this distribution.  The full text of the license may be found at
+ http://opensource.org/licenses/bsd-license.php
+ 
+ THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+ 
+ **/
+
+#include "ControlMsrE2.h"
+
+
+UINTN                     mFlags;
+
+EFI_STRING AllocateStrFromAscii (CHAR8 *s) {
+    EFI_STRING r = AllocatePool(sizeof(CHAR16) * (AsciiStrLen(s) + 1));
+    
+    CHAR16 *rp = r;
+    
+    for (CHAR8 *sp = s; *sp != 0; sp++, rp++)
+        *rp = *sp;
+    
+    *rp = 0;
+    return r;
+}
+
+UINT8 EFIAPI Char16IgnoreCaseCompare (CHAR16 a, CHAR16 b) {
```

Comment:
You could use OcStriCmp from OcStringLib.

Reply:
Will do